### ISSUE TITLE
Nested elements [Feature] 

### ISSUE THREAD

**Issue by Jesperpaulsen** [bosskmk/pluto_grid#502]
It would be cool if we could have nested elements in the table. I.e a row could have a child row, where the child has the same amount of columns as the parent. It would probably require quite some changes to the datastructure, so totally understand if it won't be prioritized/feasible.

**Comment by bosskmk** [comment]
I've tried a few times to implement that feature.

1. Grouping rows based on a specific column. (The children of the grouped row are collapsed and unfolded.)
2. A hierarchical row with children. (The children of the row are folded and unfolded.)
3. Customized child widgets in the row. (The child widget is collapsed and unfolded.)

I thought of these three functions.

I tried a few structural considerations, but I haven't found a suitable one yet.

I would like to add this feature.
I will try again at a suitable time.
Thanks for the feature suggestion.
